project/src/components/card/card.tsx

```diff
@@ -1,28 +1,88 @@
+import { CardClassNames, ImageHeight, ImageWidth } from 'const/const';
 import { Link } from 'react-router-dom';
+import { Offer } from 'types/offer';
+import { convertRatingToPercent} from 'utils/utils';
+import styles from './card.module.css';
+import PremiumMark from 'components/premium-mark/premium-mark';
+
+type CardProps = {
+  offer: Offer;
+  className: string;
+  onMouseOver?: () => void;
+  onMouseOut?:() => void;
+}
+
+export default function Card ({offer, className, onMouseOver, onMouseOut}: CardProps)
+: JSX.Element {
+  const {
+    isPremium,
+    isFavorite,
+    previewImage,
+    price,
+    rating,
+    title,
+    type,
+    id
+  } = offer;
+
+  const setFavoriteClassName = () => isFavorite
+    ? 'place-card__bookmark-button place-card__bookmark-button--active button'
+    : 'place-card__bookmark-button button';
+
+  const imageWidth = className === CardClassNames.Favorites
+    ? ImageWidth.Favorites
+    : ImageWidth.Default;
+
+  const imageHeight = className === CardClassNames.Favorites
+    ? ImageHeight.Favorites
+    : ImageHeight.Default;
+
+  let pageClass = '';
+
+  switch (className) {
+    case CardClassNames.Sities:
+      pageClass = 'cities';
+      break;
+
+    case CardClassNames.Favorites:
+      pageClass = 'favorites';
+      break;
+
+    case CardClassNames.NearPlaces:
+      pageClass = 'near-places';
+      break;
+
+    default:
+      throw new Error(`class "${className}" not recognized`);
+  }
+
+  const infoClass = className === CardClassNames.Favorites
+    ? 'favorites__card-info'
+    : '';
 
-export default function Card (): JSX.Element {
   return (
-    <article className="cities__card place-card">
-      <div className="place-card__mark">
-        <span>Premium</span>
-      </div>
-      <div className="cities__image-wrapper place-card__image-wrapper">
-        <Link to="/offer/01">
+    <article className={className} onMouseOver={onMouseOver} onMouseOut={onMouseOut}>
+
+      {isPremium ? <PremiumMark /> : ''}
```

Comment:
Conditional rendering
```suggestion
      {isPremium && <PremiumMark />}
```